datastore/longhorn.go

```diff
@@ -99,15 +99,34 @@ func (s *DataStore) filterCustomizedDefaultSettings(customizedDefaultSettings ma
 	availableCustomizedDefaultSettings := make(map[string]string)
 
 	for name, value := range customizedDefaultSettings {
-		if err := s.ValidateSetting(string(name), value); err == nil {
-			availableCustomizedDefaultSettings[name] = value
-		} else {
-			logrus.WithError(err).Errorf("invalid customized default setting %v with value %v, will continue applying other customized settings", name, value)
+		if s.isSettingValueNotExists(types.SettingName(name), value) {
+			if err := s.ValidateSetting(string(name), value); err == nil {
+				availableCustomizedDefaultSettings[name] = value
+			} else {
+				logrus.WithError(err).Warnf("invalid customized default setting %v with value %v, will continue applying other customized settings", name, value)
+			}
 		}
 	}
 	return availableCustomizedDefaultSettings
 }
 
+// isSettingValueNotExists check if the customized default setting and value not exists
+func (s *DataStore) isSettingValueNotExists(name types.SettingName, value string) bool {
```

Comment:
I feel a bit of confused because the the behavior of the helper function and its name is not quite matching.
Suggest removing the comparison of values from the function or renaming to a better function name.